tests/instrument/expr_match.ml.reference

```diff
@@ -99,3 +99,174 @@ let f x =
         (match x with
          | Foo  -> (___bisect_mark___expr_match 36; print_endline "foobar")
          | Bar  -> (___bisect_mark___expr_match 37; print_endline "barbar"))))
+let f x =
+  ___bisect_mark___expr_match 43;
+  (match x with
+   | Foo |Bar  as ___bisect_matched_value___ ->
+       ((((match ___bisect_matched_value___ with
+           | Foo  -> (___bisect_mark___expr_match 41; ())
+           | Bar  -> (___bisect_mark___expr_match 42; ())
+           | _ -> ()))[@ocaml.warning "-8-11"]);
+        print_endline "foo"))
```

Comment:
1. Very well, I will add the switch. Seems like a good idea.
2. As for the docs, I will of course mention what we do with or patterns. Which part of the explanation are you referring to for `README.md`? We have a lot of text above :) I will have a PR or PRs soon with documentation drafts, so we should be able to discuss it in more detail there. If you don't mind, I will make a note of it now, but modify the docs in that later PR.
3. Sounds good as well :)

My example with the pattern guard above was bogus, because `s ()` won't be run anyway in the second `match` expression unless `Foo` or `Bar` match first, and only one of those can match.

However, here is a slightly ridiculous example that *does* have different behavior between the two expressions. I tested in the top-level to make sure I haven't made a mistake again:

```ocaml
let s () = print_endline "foo"; false;;

match 0 with
| _ | _ when s () -> ()
| _ -> ();;

match 0 with
| _ when s () -> ()
| _ when s () -> ()
| _ -> ();;
```

The first prints `foo` once, the second twice. Granted, the first gives a warning (unused sub-pattern), but a `match` expression with redundant sub-patterns might conceivably result automatically from another preprocessor, and not be inherently suspicious.